src/Resources/ResourceManager/SdkModels/Resources/PSResource.cs

```diff
@@ -77,96 +77,92 @@ public string TagsTable
 
         public string ETag { get; set; }
 
-        public PSResource(GenericResource resource)
+        public PSResource(string id, string subscriptionId, string resourceGroupName, string name, string kind, string location, string type, Identity identity, PSObject properties, Plan plan, Sku sku, IDictionary<string, string> tags)
         {
-            this.ResourceId = resource.Id;
-            this.Id = resource.Id;
-            this.Identity = resource.Identity;
-            this.Kind = resource.Kind;
-            this.Location = resource.Location;
-            this.ManagedBy = resource.ManagedBy;
-            this.ResourceName = resource.Name;
-            this.Name = resource.Name;
-            this.Plan = resource.Plan;
-            this.Properties = ((JToken)resource.Properties).ToPsObject();
-            this.ResourceType = resource.Type;
-            this.Sku = resource.Sku;
-            this.Tags = resource.Tags;
-            this.Type = resource.Type;
-
-            var resourceIdentifier = new ResourceIdentifier(this.Id);
-            this.ParentResource = resourceIdentifier.ParentResource;
```

Comment:
ParentResource is missing.

Reply:
Good spot / apologies.  Corrected ( https://github.com/JohnLBevan/azure-powershell/blob/issue-10783-PSResource/src/Resources/ResourceManager/SdkModels/Resources/PSResource.cs#L113 )